app/routes.js

```diff
@@ -62,6 +62,21 @@ module.exports = function(app) {
             .exec(resHandler(res))
     });
 
+    // get all neighbours
+    app.get('/api/sample/neighbours/:id', function(req, res) {
```

Comment:
This is basically exactly what I'm talking about with anonymous functions. This should read:

```javascript
app.get('api/sample/neighbours/:id', get_neighbours(req.params.id, res))
```

Later, you can do something similar with getting the thumbnails.